serviceconnector/src/createServiceConnector/ICreateLinkerContext.ts

```diff
@@ -0,0 +1,24 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.md in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+import { AuthInfoBase, KnownClientType, LinkerResource } from "@azure/arm-servicelinker";
+import { IStorageAccountWizardContext } from "@microsoft/vscode-azext-azureutils";
+import { ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { TargetServiceType } from "../../constants";
+
+
+export interface ICreateLinkerContext extends ISubscriptionActionContext, IStorageAccountWizardContext {
```

Comment:
Do we need to add any support for `AzureSubscription`?

@alexweininger 
I noticed in V2 migration of Container Apps that we introduced an `AzureSubscription` to the context.  Is this eventually supposed to replace `ISubscriptionActionContext`?  Do we need to do add additional support here?

Reply:
TLDR I think that this is fine.

As far as I know, they're different things and an `ISubscriptionActionContext` can be created from an `AzureSubscription`, but not vice-versa.  `ISubscriptionActionContext` basically takes a subscription object (I can't remember what the T1 version was called, but it was the equivalent of `AzureSubscription` and combines it with an `IActionContext`.

Reply:
There is quite a bit of context needed here...

Brandon and Nathan have been working over the past year (maybe two years) or so to migrate from Track 1 Azure SDKs to Track 2 Azure SDKs. One of the significant differences between T1 and T2 is how they handle authentication (switching to @azure/identity from @azure/ms-rest-nodeauth).

`AzureSubscription` is the new gold standard for representing a subscription within the extensions, and is aligned with the Track 2 authentication formats. It just so happened that it was convenient to introduced this new subscription format as part of the Azure Resources API v2 rewrite.

For `ISubscriptionActionContext`, I think we can change it out later on for a simple interface that just has an `AzureSubscription` on it. I believe this will need to happen when the extensions upgrade to use the v2 shared packages which are fully switched to Track 2 SDKs. It _should_ be a fairly easy thing.